analytics-integrations/uxcam/src/main/java/com/segment/analytics/internal/integrations/UXCamIntegration.java

```diff
@@ -0,0 +1,39 @@
+package com.segment.analytics.internal.integrations;
+
+import android.app.Activity;
+import android.os.Bundle;
+
+import com.segment.analytics.Analytics;
+import com.segment.analytics.ValueMap;
+import com.segment.analytics.internal.AbstractIntegration;
+import com.uxcam.UXCam;
+/**
+ * UXCam allows you to eliminate customer struggle and improve user experience by capturing
+ * and visualizing screen video and user interaction data.
+ *
+ * @see <a href="http://www.uxcam.com/">UXCam</a>
+ * @see <a href="https://dashboard.uxcam.com/docs#/android">UXCam Android SDK</a>
+ */
+public class UXCamIntegration extends AbstractIntegration<UXCam> {
```

Comment:
Hi Prateek,

Please find the answers to your questions inline.


*Why is Mixpanel a dependency in testCompile?*

We have a feature to tag Mixpanel with UXCam video URL if mixpanel SDK is
present. Since the test cases failed due to not finding Mixpanel SDK, we
had to include Mixpanel SDK as a dependency in testCompile.

*If Play Services is required, it should be regular compile dependency.*

UXCam has provided dependency on play-services for taking MapFragment
screenshots. Same as above, test cases were failed and hence play-services
was included as testCompile. UXCam doesn’t depend on play-services.

*Don't you want to be collecting user data here? At least track and
identify, probably screen as well.*

We are not using this data provided by segment at the moment and hence have
not collected them yet. It's something we are considering using on the next
push though, hope that would be ok ?

Let me know if you have more questions.
ᐧ


On Wed, Jun 3, 2015 at 11:19 AM, Prateek Srivastava <
notifications@github.com> wrote:

> In
> analytics-integrations/uxcam/src/main/java/com/segment/analytics/internal/integrations/UXCamIntegration.java
> <https://github.com/segmentio/analytics-android/pull/289#discussion_r31653071>
> :
>
> > +
> > +import android.app.Activity;
> > +import android.os.Bundle;
> > +
> > +import com.segment.analytics.Analytics;
> > +import com.segment.analytics.ValueMap;
> > +import com.segment.analytics.internal.AbstractIntegration;
> > +import com.uxcam.UXCam;
> > +/**
> > + * UXCam allows you to eliminate customer struggle and improve user experience by capturing
> > + * and visualizing screen video and user interaction data.
> > + *
> > + * @see <a href="http://www.uxcam.com/">UXCam</a>
> > + * @see <a href="https://dashboard.uxcam.com/docs#/android">UXCam Android SDK</a>
> > + */
> > +public class UXCamIntegration extends AbstractIntegration<UXCam> {
>
> Don't you want to be collecting user data here? At least track and
> identify, probably screen as well.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/segmentio/analytics-android/pull/289/files#r31653071>.
>



-- 
-- 
Kishan Gupta
CEO | UXCam
Understand your users w. UXCam <http://www.uxcam.com>